src/modules/hip/kernel/mel_filter_bank.hpp

```diff
@@ -0,0 +1,153 @@
+{
+    dstVal = 0;
+    //start and end FFT bin indices for the current mel bin
+    int fftbin = intervals[melBin];
+    int fftBinEnd = intervals[melBin + 1];
+
+    float *srcPtrTemp = srcPtr + fftbin * fftStrides.x + fftStrides.y;
+    // Process the first interval of FFT bins, applying the weights up
+    for (; fftbin < fftBinEnd; fftbin++, srcPtrTemp += fftStrides.x) 
+    {
+        float weightUp = 1.0f - weightsDown[fftbin];
+        weightUp *= normFactor;
+        dstVal += *srcPtrTemp * weightUp;
+    }
+
+    fftBinEnd = intervals[melBin + 2];    // Update the end FFT bin index for the next interval
+    srcPtrTemp = srcPtr + fftbin * fftStrides.x + fftStrides.y;
+
+    // Process the second interval of FFT bins, applying the weights down
+    for (; fftbin < fftBinEnd; fftbin++, srcPtrTemp += fftStrides.x) 
+    {
+        float weightDown = weightsDown[fftbin];
+        weightDown *= normFactor;
+        dstVal += *srcPtrTemp * weightDown;
+    }
+}
+
+__global__ void mel_filter_bank_tensor(float *srcPtr,
+                                       uint2 srcStridesNH,
+                                       float *dstPtr,
+                                       uint2 dstStridesNH,
+                                       int *srcDimsTensor,
+                                       int numFilter,
+                                       bool normalize,
+                                       float *normFactors,
+                                       float *weightsDown,
+                                       int *intervals)
+{
+    int id_x = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    if (id_x >= srcDimsTensor[id_z * 2 + 1] || id_y >= numFilter)
+        return;
+
+    uint dstIdx = id_z * dstStridesNH.x + id_y * dstStridesNH.y + id_x;
+    uint srcIdx = id_z * srcStridesNH.x;
+
+    float normFactor = (normalize) ? normFactors[id_y] : 1;
+    compute_mel(srcPtr + srcIdx, id_y, weightsDown, intervals, make_int2(srcStridesNH.y, id_x), normFactor, dstPtr[dstIdx]);
+}
+
+RppStatus hip_exec_mel_filter_bank_tensor(Rpp32f *srcPtr,
+                                          RpptDescPtr srcDescPtr,
+                                          Rpp32f *dstPtr,
+                                          RpptDescPtr dstDescPtr,
+                                          Rpp32s* srcDimsTensor,
+                                          Rpp32f maxFreqVal,
+                                          Rpp32f minFreqVal,
+                                          RpptMelScaleFormula melFormula,
+                                          Rpp32s numFilter,
+                                          Rpp32f sampleRate,
+                                          bool normalize,
+                                          rpp::Handle& handle)
+{
+    BaseMelScale *melScalePtr;
+    switch (melFormula)
```

Comment:
For this initial setup from L70-L132 (before you set globalThreads), can we have some simple one liner comments for each logically separated code block as to what you are doing or setting up

Reply:
Done